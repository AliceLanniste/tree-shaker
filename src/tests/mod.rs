fn tree_shake(input: &str) {
  let result = crate::tree_shake(input);
  let output = result.codegen_return.source_text;
  insta::assert_snapshot!(output);
}

#[test]
fn test_1() {
  tree_shake(
    r#"
      export let a = 1 && 2;
      export { b };
      let c = 3;
      let { ["b"]: b, d } = { b: 2, d: effect };
      "#,
  );
}
