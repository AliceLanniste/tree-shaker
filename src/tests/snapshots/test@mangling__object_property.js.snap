---
source: src/tests/mod.rs
expression: tree_shake(input)
input_file: src/tests/fixtures/mangling/object_property.js
---
export function plain_object(aaa, bbb) {
	const obj = {
		a: aaa,
		b: bbb
	};
	console.log(obj.a);
	console.log(obj.b);
}
export function computed_property(unknown, aaa, bbb) {
	const key = unknown ? "a" : "b";
	const obj = {
		a: aaa,
		b: bbb
	};
	console.log(obj[key]);
}
export function property_via_destructuring(aaa, bbb) {
	const { a: foo, b: bar } = {
		a: aaa,
		b: bbb
	};
	console.log(foo);
	console.log(bar);
}
export function with_rest(__unused_7A33, bbb) {
	const { a: __unused_EBAD,...rest } = { bar: bbb };
	console.log(rest.bar);
}
export function dynamic_destructuring(unknown, aaa, bbb) {
	const { [unknown ? "a" : "b"]: value } = {
		a: aaa,
		b: bbb
	};
	console.log(value);
}
